book/src/kernel/linux-compatibility/limitations-on-system-calls/system-call-matching-language.md

```diff
@@ -277,20 +339,27 @@ Non‑terminals are in angle brackets, terminals in quotes.
 <param-list>     ::= '..'
                    | <param> { ',' <param> } [ ',' '..' ]
 <param>          ::= <identifier> '=' <expr>
+                   | <identifier> '=' <struct-pattern>
+                   | <identifier> '=' <array-pattern>
                    | <identifier>
 
-<expr>           ::= <expr> '|' <expr>
-                   | <term>
+<expr>           ::= <term> { '|' <term> }
 <term>           ::= <identifier>
                    | '<' <identifier> '>'
 
-<array>          ::= '[' '<' <identifier> '>' ']'  
+<array-pattern>  ::= '[' <array-element> { ',' <array-element> } ']'
+<array-element>  ::= '<' <identifier> '>'
+                   | <struct-pattern>
+                   | <array-pattern>
+
+<struct-rule>    ::= 'struct' <identifier> '=' <struct-pattern>
+<struct-pattern> ::= '{' <field-list> [ ',' '..' ] '}'
 
-<struct-rule>    ::= 'struct' <identifier> '=' '{' <field-list> [ ',' '..' ] '}'
 <field-list>     ::= <field> { ',' <field> }
```

Comment:
I think the blank line above should be removed as `<field-list>` is only used as part of `<struct-rule>`.